src/lib/slot.c

```diff
@@ -118,23 +118,78 @@ CK_RV slot_mechanism_info_get (CK_SLOT_ID slot_id, CK_MECHANISM_TYPE type, CK_ME
         return CKR_SLOT_ID_INVALID;
     }
 
-    // TODO support more of these and check with the TPM for sizes.
+    /* TODO pull these from TPM, currently they match the simulator */
```

Comment:
Let's not do this halfway, lets go to the TPM and run some getcap commands and actually look at what is supported. Not everything actually supports AES.

Reply:
I would suggest one step at a time. I want to split this up in reviewable chunks.
This definitely improves the situation, as the fields are atleast populated, with sane values.
The real fix comes then when #198 gets closed.

For most of these details, there is unfortunately no easy getcap (how to retrieve rsa keysizes?), just a tpm_testparms where you can 'test' whether the specified template is supported.
This is a whole new chunk of functionality, which I then can plug in here.

For AES, yes correct, but if it's not supported at all, the mechanisms will not be in the mechanism_info_list. (All applications I'm aware of query the provider if a mechanissm is supported at all and then for the details. Or they don't query anything)

Reply:
I guess since it's only related to #198 and it leaves that open, it will serve as a reminder to ensure this doesn't last forever.